internal/quantitative/local_engine.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2022 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package quantitative
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/corazawaf/coraza/v3"
+	"github.com/corazawaf/coraza/v3/types"
+	"github.com/rs/zerolog/log"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"text/template"
+)
+
+const (
+	defaultPrefix     = "."
+	testingConfigTmpl = `SecRuleEngine On
+SecRequestBodyAccess On
+SecRule REQUEST_HEADERS:Content-Type "^(?:application(?:/soap\+|/)|text/)xml" \
+     "id:'200000',phase:1,t:none,t:lowercase,pass,nolog,ctl:requestBodyProcessor=XML"
+SecRule REQUEST_HEADERS:Content-Type "^application/json" \
+     "id:'200001',phase:1,t:none,t:lowercase,pass,nolog,ctl:requestBodyProcessor=JSON"
+SecRule REQUEST_HEADERS:Content-Type "^application/[a-z0-9.-]+[+]json" \
+     "id:'200006',phase:1,t:none,t:lowercase,pass,nolog,ctl:requestBodyProcessor=JSON"
+SecRequestBodyLimit 13107200
+SecRequestBodyInMemoryLimit 131072
+SecRequestBodyLimitAction Reject
+SecRule REQBODY_ERROR "!@eq 0" \
+    "id:'200002', phase:2,t:none,log,deny,status:400,msg:'Failed to parse request body.',logdata:'%{reqbody_error_msg}',severity:2"
+SecRule MULTIPART_STRICT_ERROR "!@eq 0" \
+    "id:'200003',phase:2,t:none,log,deny,status:400, \
+    msg:'Multipart request body failed strict validation."
+SecResponseBodyAccess On
+SecResponseBodyMimeType text/plain text/html text/xml
+SecResponseBodyLimit 524288
+SecResponseBodyLimitAction ProcessPartial
+SecDataDir /tmp/
+SecAuditEngine RelevantOnly
+SecAuditLogRelevantStatus "^(?:(5|4)(0|1)[0-9])$"
+SecAuditLogParts ABIJDEFHZ
+SecAuditLogType Serial
+SecAction \
+    "id:900000,\
+    phase:1,\
+    pass,\
+    t:none,\
+    nolog,\
+    tag:'OWASP_CRS',\
+    ver:'OWASP_CRS/4.7.0-dev',\
+    setvar:tx.blocking_paranoia_level={{ .ParanoiaLevel }}"
+`
+)
+
+type LocalEngine struct {
+	waf coraza.WAF
+}
+
+// NewEngine creates a new engine to test payloads
+func NewEngine(prefix string, paranoia int) *LocalEngine {
+	eng := &LocalEngine{
+		waf: crsWAF(prefix, paranoia),
+	}
+	return eng
+}
+
+// CRSCall benchmarks the CRS WAF with a GET request
+// payload: the string to be passed in the request body
+// returns the status of the transaction and a map of the matched rules with their IDs and the data that matched.
```

Comment:
Ideally, the return value should be another interface backed by a struct. May be premature.

```suggestion
// returns the status of the HTTP response and a map of the matched rules with their IDs and the data that matched.
```